docs: Add CMake version requirement for Linux source builds

WangMiao02 opened the pull request:
Specify that CMake version 3.17 or higher is required for building BMF from source on Linux to ensure successful compilation.

WangMiao02 committed:
docs: Add CMake version requirement for Linux source builds

Specify that CMake version 3.17 or higher is required for building BMF from source on Linux to ensure successful compilation.

JackLau1222 commented on content/zh/docs/bmf/getting_started_yourself/install/_index.md:
```diff
@@ -314,6 +314,9 @@ BMF 支持在以下三个平台编译和构建：Linux、Windows 和 Mac。您
 ### Linux
 
+**重要提示：CMake 版本要求**
+在 Linux 环境下从源代码构建 BMF 时，请确保您的系统中安装的 CMake 版本 **不低于 3.17**。较低版本的 CMake 可能会导致构建失败或出现不可预知的错误。您可以通过运行 `cmake --version` 来检查当前的 CMake 版本。如果版本过低，请先升级 CMake。
```
Hi, I wanna know how are you getting this specific version `3.17` ?

WangMiao02 replied:
Hi @JackLau1222,

Thanks for asking! The requirement for CMake version 3.17 (or higher) comes directly from the BMF framework's `CMakeLists.txt` file.

When I attempted to build BMF from source using CMake version 3.16.3, the configuration process failed, with the following specific error:
![_cgi-bin_mmwebwx-bin_webwxgetmsgimg__ MsgID=2087615060352367289 skey=@crypt_464b150b_dacddea82203961ea7587939bcddd736 mmweb_appid=wx_webfilehelper](https://github.com/user-attachments/assets/c8c56e3c-da54-469e-95ae-26e2492ad183)


As you can see, the `cmake_minimum_required` directive on line 33 of the project's main `CMakeLists.txt` explicitly specifies that CMake 3.17 or a newer version is necessary for the build system to proceed. Using an older version like 3.16.3 prevents the configuration pTM from completing, which in turn stops the entire build process.

Therefore, to ensure a successful source build on Linux, upgrading CMake to version 3.17 or later is a prerequisite.

JackLau1222 replied:
Thanks for explanation, but i think it's worth to improve that version 3.17 is required when enabling CUDA, while the default only requires 3.5

WangMiao02 replied:
Hi @JackLau1222,

Thank you for your careful review and correction. You are absolutely right. I apologize for any confusion in my previous explanation.

After carefully reviewing the CMakeLists.txt file in the bmf/hmp directory again, I noticed that:

1. The base CMake version requirement is 3.5:
```cmake
cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
```

2. The version requirement is elevated to 3.17 only when CUDA support is enabled:
```cmake
if (HMP_ENABLE_CUDA)
    # For FindCUDAToolkit support
    cmake_minimum_required(VERSION 3.17 FATAL_ERROR)
endif()
```

I will update the documentation to reflect this more accurately - making it clear that CMake 3.17+ is only required when building with CUDA support (which is enabled by default), while the base requirement is CMake 3.5.

Thank you for helping maintain the accuracy of our documentation!

WangMiao02 committed:
docs: clarify CMake version requirements for BMF source build

Update CMake version requirements in the installation documentation to be more accurate:

- Clarify that the base CMake version requirement is 3.5
- Specify that CMake 3.17+ is only required when CUDA support is enabled (default)
- Add this distinction to prevent confusion for users building without CUDA support

This change makes the documentation more precise and helps users better understand
the build requirements based on their specific needs.

JackLau1222 commented on content/zh/docs/bmf/getting_started_yourself/install/_index.md:
```diff
@@ -314,6 +314,12 @@ BMF 支持在以下三个平台编译和构建：Linux、Windows 和 Mac。您
+在 Linux 环境下从源代码构建 BMF 时：
+- 基础版本要求：CMake 3.5 或更高版本
+- 如果需要 CUDA 支持，则要求 CMake 3.17 或更高版本
+- 您可以通过运行 `cmake --version` 来检查当前的 CMake 版本。如果版本不满足要求，请先升级 CMake。
```
Maybe put it in top of Linux is more matched because the requirements need to be required cross platform

WangMiao02 replied:
Thank you for your valuable suggestion! I have updated the documentation to move the CMake version requirements to the top of the Linux section, as you recommended. This change ensures that the requirements are clearly stated and applicable across platforms.

JackLau1222 commented on content/zh/docs/bmf/getting_started_yourself/install/_index.md:
and if you could finish the English version also?

WangMiao02 replied:
I wanted to highlight that the updates to both the Chinese and English versions of the documentation were made thanks to your reminder. Your input was invaluable in ensuring that the CMake version requirements are clearly stated.

JackLau1222 replied:
I think you can put the English and Chinese changes in one commit rather than submit another PR

WangMiao02 committed:
docs: move CMake version requirements to cross-platform section

Improve documentation structure by relocating CMake version requirements:

- Move CMake version requirements from Linux-specific section to the top of 
  "Building from Source" section
- This change reflects that CMake requirements are cross-platform
- Requirements remain the same:
  * Base requirement: CMake 3.5+
  * CUDA support (enabled by default): CMake 3.17+
- No content changes, only structural improvement for better readability
  and maintenance

This reorganization helps users understand the build requirements before
starting the build process on any platform.

JackLau1222 commented on content/zh/docs/bmf/getting_started_yourself/install/_index.md:
```diff
@@ -315,10 +315,11 @@ BMF 支持在以下三个平台编译和构建：Linux、Windows 和 Mac。您
 ### Linux
```
put the changes in top of this `### Linux`, your English version works well.

WangMiao02 committed:
Apply to .gitignore docs: add CMake version requirements for source builds

Add CMake version requirements to the "Build from Source" section of installation documentation.

WangMiao02 committed:
docs: Add CMake version requirement for Linux source builds(EN)

Add CMake version requirements to the "Build from Source" section of installation documentation.

WangMiao02 committed:
Update _index.md

Fix a small error.
